### PR TITLE
Create Contributing.md issue #33

### LINKED ISSUE #33: Add Contributing.md
Reference Link: https://github.com/github/docs/blob/main/CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by Devansh-Singh-Parmar**
please let me know what to add or what to remove!

### CODE REVIEW

File: Contributing.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
+# Contributing to DeliverEase
+
+Welcome to DeliverEase! We are excited to have you contribute to our project. DeliverEase is made using Django (Python) for the backend and HTML/CSS for the frontend. Your contributions can help us improve and enhance the platform, empowering delivery partners and connecting communities even more effectively.
```
Comment (SID262000) [inline Contributing.md]: I wouldn't recommend adding the tech stack explicitly as that is really volatile with time.
cc: @Devansh-Singh-Parmar

File: Contributing.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+## How to Contribute
+
+To contribute to DeliverEase, please follow these guidelines:
+
+1. Fork the repository to your GitHub account.
+
```
Comment (SID262000) [inline Contributing.md:10]: Add cloning step.
@Devansh-Singh-Parmar

File: Contributing.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+2. Create a new branch from the `main` branch. Use a descriptive name for your branch that reflects the nature of your contribution.
+
+3. Make your changes to the codebase or documentation in your branch. Be sure to follow the coding style and conventions used in the project.
+
+4. Write clear and concise commit messages for your changes.
```
Comment (SID262000) [inline Contributing.md]: Add a instruction to attach issue number while committing to track it as part of the Issue.
cc: @Devansh-Singh-Parmar

File: Contributing.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+5. Test your changes thoroughly to ensure they work as expected.
+
+6. Update or create new tests where applicable.
+
+7. Ensure that all existing tests pass successfully before submitting your changes.
+
+8. Push your branch to your forked repository.
+
+9. Create a pull request (PR) from your branch to the `main` branch of the main repository.
```
Comment (SID262000) [inline Contributing.md]: This sounds little confusing, we need to clearly indicate the upstream & origin repos here.
cc: @Devansh-Singh-Parmar

File: Contributing.md [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+10. Provide a detailed description of your changes in the pull request, including the problem you are solving, the approach you took, and any relevant information for reviewers.
```
Comment (SID262000) [inline Contributing.md]: Attach a section to include the Issue number as well while creating the PR.
[Reference](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#:~:text=The%20issue%20and%20pull%20request,to%20link%20to%20an%20issue.)
cc: @Devansh-Singh-Parmar

File: Contributing.md [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+
+11. Be responsive to feedback and comments on your pull request. If requested changes are needed, make the necessary adjustments.
+
+12. Once your pull request is approved and meets the project's quality standards, it will be merged into the main repository.
+
+## Code Style and Guidelines
+
+To maintain a consistent codebase, we follow specific code style and guidelines. Please adhere to the following conventions when contributing:
```
Comment (SID262000) [inline Contributing.md:41]: Attaching a link to the best practices docs officially would make more sense here.
cc: @Devansh-Singh-Parmar

File: Contributing.md [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+
+If you come across any issues, bugs, or have suggestions for improvements, please report them using GitHub Issues. When reporting issues, provide as much relevant information as possible, including steps to reproduce the problem, your operating system, browser details, etc.
+
+## Code of Conduct
+
+We expect all contributors to adhere to our Code of Conduct, which promotes a welcoming and inclusive community. Please be respectful and considerate of others' opinions and contributions.
+
+## License
+
+By contributing to DeliverEase, you agree that your contributions will be licensed under the [MIT License](https://opensource.org/licenses/MIT).
```
Comment (SID262000) [inline Contributing.md]: This might need to be updated in case we decide to go with any other License.
cc: @Devansh-Singh-Parmar @N3v1

**Review by SID262000 — CHANGES_REQUESTED** [review]
I've requested a bunch of changes at a couple of steps, feel free to review the same and implement it, Thanks
cc: @Devansh-Singh-Parmar

**Review by N3v1 — CHANGES_REQUESTED** [review]
I would also add links to the Github documentations at the section "How to contribiute". Example: 
```md
1. [**Fork**](https://help.github.com/articles/fork-a-repo/) this project
2. [**Clone**](https://help.github.com/articles/fork-a-repo/#step-2-create-a-local-clone-of-your-fork) your forked version
3. **Make changes**
4. **Create a [branch](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/ about-branches#working-with-branches)**
5. **[Commit](https://help.github.com/articles/adding-a-file-to-a-repository-using-the-command-line/) your changes** *(write a short descriptive message of what you have done)*
6. [**Push**](https://help.github.com/articles/pushing-to-a-remote/) your changes to your forked version
7. Go to the original project on GitHub & **Create a [Pull Request](https://help.github.com/articles/about-pull-requests/)**
```

File: Contributing.md [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+
+## Comment on Issues
+
+You can actively participate by commenting on the open issues. If you think you can solve a particular issue, comment "Please assign this issue to me," and one of the maintainers will assign the issue to you. Once assigned, you can start working on the problem and create a pull request once your solution is ready.
+
+## Reporting Issues
+
+If you come across any issues, bugs, or have suggestions for improvements, please report them using GitHub Issues. When reporting issues, provide as much relevant information as possible, including steps to reproduce the problem, your operating system, browser details, etc.
+
+## Code of Conduct
```
Comment (N3v1) [inline Contributing.md:57]: Hi @Devansh-Singh-Parmar 👋,
here you have to link the CODE_OF_CONDUCT.md file. 

Heres the correct code for this section:
```md
We expect all contributors to adhere to our Code of Conduct, which promotes a welcoming and inclusive community. Please be respectful and considerate of others' opinions and contributions.
See [`**CODE_OF_CONDUCT**`](CODE_OF_CONDUCT.md)
```

If your Contributing.md is in a different folder, you would have to change the link to CODE_OF_CONDUCT.md

File: Contributing.md (hunk 7 above)
Comment (N3v1) [reply Contributing.md]: I would separate the text from the license text. Just leave three
Free lines between the two texts and in the second line write --- , or indicate the "Thank you ..." text as a heading with ##.

**Review by N3v1 — CHANGES_REQUESTED** [review]
I would separate the text from the license text. Just leave three
Free lines between the two texts and in the second line write --- , or indicate the "Thank you ..." text as a heading with ##.

File: Contributing.md [hunk 9]
```diff
@@ -0,0 +1,59 @@
+# Contributing to DeliverEase
```
Comment (N3v1) [inline Contributing.md]: Maybe you can bold or italicize important words or phrases. That would increase the readability tremendously

File: Contributing.md (hunk 2 above)
Comment (Devansh-Singh-Parmar) [reply Contributing.md:10]: Please explain this better.

Comment (N3v1) [reply Contributing.md:10]: To contribute, please ensure that you have added the cloning step. The contributors fork the repository and then clone the fork onto their computers. After this, they create a new branch.

You just have to add this step:
```md
2. Clone your fork onto your computer
```

File: Contributing.md (hunk 6 above)
Comment (Devansh-Singh-Parmar) [reply Contributing.md:41]: Requesting a reference

File: Contributing.md (hunk 7 above)
Comment (Devansh-Singh-Parmar) [reply Contributing.md]: I am removing License section for now, please choose the license.

cc: @SID262000 @N3v1

**Comment by Devansh-Singh-Parmar** [comment]
Please let me know any other changes!

**Review by N3v1 — CHANGES_REQUESTED** [review]
Maybe you can bolden some important words or phrases @Devansh-Singh-Parmar 
And link to the CODE_OF_CONDUCT.md

**Comment by Devansh-Singh-Parmar** [comment]
Added the Link to the code_of_conduct.md already, please check, and will bold the phrased or important words in next commit 

cc: @N3v1

**Comment by N3v1** [comment]
> Added the Link to the code_of_conduct.md already, please check, and will bold the phrased or important words in next commit
> 
> cc: @N3v1

Oh yes, you're right. My mistake 🙈 @Devansh-Singh-Parmar 
Sorry

File: readme.md [hunk 10]
```diff
@@ -20,7 +20,7 @@
 - [Contributing](#contributing)
 - [Feedback](#feedback)
 - [License](#license)
-- [How you can reach us]()
+- [How you can reach us](#social-media)
```
Comment (N3v1) [inline readme.md:23]: Oh I forgot it, thank you @Devansh-Singh-Parmar 🙏

**Review by N3v1 — CHANGES_REQUESTED** [review]
Maybe you could add a table of content, if you want to

**Comment by Devansh-Singh-Parmar** [comment]
@N3v1  What do you mean by that table of content?

also, asking for a final review from @SID262000  before making a final PR

**Comment by AryanGupta0007** [comment]
> Maybe you could add a table of content, if you want to 

Table of content is not needed as the contributors should go through each and every guideline. Hence i feel it's not necessary to add that 
Cc: @Devansh-Singh-Parmar @SID262000

**Comment by Devansh-Singh-Parmar** [comment]
Asking again for a review from 
cc : @SID262000

**Comment by SID262000** [comment]
Already answered in preceding PR, Thanks

**Pull request closed without merging**